src/codegen/codegen_c_visitor.cpp

```diff
@@ -2949,34 +3018,41 @@ void CodegenCVisitor::print_thread_memory_callbacks() {
 }
 
 
-void CodegenCVisitor::print_mechanism_range_var_structure() {
+void CodegenCVisitor::print_mechanism_range_var_structure(bool print_initialisers) {
+    auto const value_initialise = print_initialisers ? "{}" : "";
     auto float_type = default_float_data_type();
     auto int_type = default_int_data_type();
     printer->add_newline(2);
-    printer->add_line("/** all mechanism instance variables */");
-    printer->start_block(fmt::format("struct {} ", instance_struct()));
+    printer->add_line("/** all mechanism instance variables and global variables */");
+    printer->fmt_start_block("struct {} ", instance_struct());
+    // TODO dynamically [don't] include this depending on whether it's used
+    printer->fmt_line("double celsius{};", print_initialisers ? "{celsius}" : "");
```

Comment:
I think celsius should be in the symbol table and there is a "read/write" count per symbol that indicates if a variable is used in the mod file or not.

Reply:
Done in 07086dfa38e308699c86212cf98d5324217995bc.